crates/bevy_render/src/camera/camera.rs

```diff
@@ -203,3 +209,109 @@ pub fn camera_system<T: CameraProjection + Component>(
         }
     }
 }
+
+pub struct CameraTypePlugin<T: Component + Default>(PhantomData<T>);
+
+impl<T: Component + Default> Default for CameraTypePlugin<T> {
+    fn default() -> Self {
+        Self(Default::default())
+    }
+}
+
+impl<T: Component + Default> Plugin for CameraTypePlugin<T> {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<ActiveCamera<T>>()
+            .add_startup_system_to_stage(StartupStage::PostStartup, set_active_camera::<T>)
+            .add_system_to_stage(CoreStage::PostUpdate, set_active_camera::<T>);
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app.add_system_to_stage(RenderStage::Extract, extract_cameras::<T>);
+        }
+    }
+}
+
+/// The canonical source of the "active camera" of the given camera type `T`.
+#[derive(Debug)]
+pub struct ActiveCamera<T: Component> {
+    camera: Option<Entity>,
+    marker: PhantomData<T>,
+}
+
+impl<T: Component> Default for ActiveCamera<T> {
+    fn default() -> Self {
+        Self {
+            camera: Default::default(),
+            marker: Default::default(),
+        }
+    }
+}
+
+impl<T: Component> Clone for ActiveCamera<T> {
+    fn clone(&self) -> Self {
+        Self {
+            camera: self.camera,
+            marker: self.marker,
+        }
+    }
+}
+
+impl<T: Component> ActiveCamera<T> {
+    /// Sets the active camera to the given `camera` entity.
+    pub fn set(&mut self, camera: Entity) {
+        self.camera = Some(camera);
+    }
+
+    /// Returns the active camera, if it exists.
+    pub fn get(&self) -> Option<Entity> {
+        self.camera
+    }
+}
+
+pub fn set_active_camera<T: Component>(
+    mut active_camera: ResMut<ActiveCamera<T>>,
+    cameras: Query<Entity, With<T>>,
+) {
+    if active_camera.get().is_some() {
+        return;
+    }
+
+    if let Some(camera) = cameras.iter().next() {
+        active_camera.camera = Some(camera);
+    }
+}
+
+#[derive(Component, Debug)]
+pub struct ExtractedCamera {
+    pub target: RenderTarget,
+    pub physical_size: Option<UVec2>,
+}
+
+pub fn extract_cameras<M: Component + Default>(
+    mut commands: Commands,
+    windows: Res<Windows>,
+    images: Res<Assets<Image>>,
+    active_camera: Res<ActiveCamera<M>>,
+    query: Query<(Entity, &Camera, &GlobalTransform, &VisibleEntities), With<M>>,
+) {
+    for (entity, camera, transform, visible_entities) in query.iter() {
+        if let Some(size) = camera.target.get_physical_size(&windows, &images) {
+            commands.get_or_spawn(entity).insert_bundle((
```

Comment:
I understand there's likely only a handful of these cameras, but is it possible to switch this to use the `insert_or_spawn_batch` pattern for extraction used in #4179? This could speed notably speed up extraction here.

Reply:
Hmm wait this should actually just extract the active camera. Might as well make this O(1).